docs/source/udf/profiling.rst

```diff
@@ -0,0 +1,180 @@
+.. _`udf profiling`:
+
+.. testsetup:: *
+
+    from libertem.api import Context
+
+
+Profiling UDFs
+==============
+
+To bring your UDF to production, it may be necessary to look into the performance and
+efficiency of your function. There may be some low-hanging fruit to optimize away, but
+they only really become visible once you profile your code. Just looking at code and guessing
+which parts are expensive can be misleading and cause you to spend time optimizing the wrong part
+of your code. Always measure!
+
+Profiling means instrumenting the execution of a program and finding out which parts
+of your code use how much of a given resource (for example, CPU time, or memory). 
+
+Prerequisite: inline executor
+-----------------------------
+
+By default, your UDF will be run in a multi-process or multi-threaded environment. This makes
+profiling a challenge. But as most of the time, improving the execution time of your UDF in a
+single-threaded executor also improves the respective multi-process execution time, we can run
```

Comment:
This sentence should logically come after the next IMO.

Reply:
What about

In order to use the default Python profiling mechanisms, all parts of the UDF should be executed in the same single thread. LiberTEM provides what it calls :class:`~libertem.executor.inline.InlineJobExecutor` for this purpose. Improving the execution time in a single-threaded executor will almost always improve the multi-process execution time. Nevertheless, you should confirm the impact of changes in a multi-processing or multi-threaded environment as well.

Reply:
I tried to merge this and the previous comment into something cohesive, let me know what you think

Reply:
Nice, I like it! :-)